lib/node_modules/@stdlib/utils/parse-dsv/lib/main.js

```diff
@@ -0,0 +1,369 @@
+				this.state = STATE_ENUM.END;
+				return;
+			}
+			break;
+		case STATE_ENUM.END:
+			this.emit('error', new Error( 'Parser is closed' ));
+			return;
+		default:
+			this.emit('error', new Error( 'Unknown state: ' + this.state ));
+			this.state = STATE_ENUM.END;
+			return;
+		}
+	}
+
+	/* eslint-enable no-fallthrough */
+}
+
+/**
+* Attempts to generate tokens from the current buffer.
+*
+* @private
+* @returns {Array} an array of tokens
+*/
+function readNextTokens() {
+	var matched;
+	var tokens = [];
+	var char;
+	var i;
+	while ( this.buffer.length > 0 ) {
+		matched = true;
+		char = this.buffer[ 0 ];
+		if ( char === this.DELIM[ 0 ] ) {
+			for ( i = 1;
+				i < this.buffer.length && i < this.DELIM.length;
+				i++ ) {
+				if ( this.buffer[ i ] !== this.DELIM[ i ] ) { // A mismatch
+					matched = false;
+					break;
+				}
+			}
+			if ( matched ) { // Everything matched
+				if ( this.buffer.length >= this.DELIM.length ) { // Entire delim was matched
+					this.buffer.splice( 0, this.DELIM.length ); // Remove the characters
+					tokens.push({
+						'type': TOKEN_ENUM.DELIM,
+						'data': this.DELIM
+					});
+					continue;
+				} else { // Not enough to determine if delim or not
+					return tokens;
+				}
+			}
+			// If mismatch, fall through and interpret char as normal
+		}
+		switch ( char ) {
+		case CR:
+			if ( this.buffer.length < 2 ) {
+				return tokens; // Need at least 1 more byte to determine token
+			}
+			if ( this.buffer[1] === LF ) {
+				this.buffer.splice( 0, 2 ); // Remove CRLF from buffer;
+				tokens.push({
+					'type': TOKEN_ENUM.CRLF,
+					'data': CRLF
+				});
+			} else {
+				this.buffer.shift();
+				tokens.push({
+					'type': TOKEN_ENUM.CR,
+					'data': CR
+				});
+			}
+			break;
+		case LF:
+			this.buffer.shift();
+			tokens.push({
+				'type': TOKEN_ENUM.LF,
+				'data': LF
+			});
+			break;
+		case this.QUOTECHAR:
+			this.buffer.shift();
+			tokens.push({
+				'type': TOKEN_ENUM.QUOTE,
+				'data': this.QUOTECHAR
+			});
+			break;
+		case null:
+			this.buffer.shift();
+			tokens.push({
+				'type': TOKEN_ENUM.ENDOFSTREAM,
+				'data': null
+			});
+			return tokens;
+		default:
+			this.buffer.shift();
+			tokens.push({
+				'type': TOKEN_ENUM.TEXTDATA,
+				'data': char
+			});
+		}
+	}
+	return tokens;
+}
+
+/**
+* When called, the function attempts to parse the input and update dsv state.
+*
+* @name incrparse
+* @memberof parseDsvStream.prototype
+* @param {Buffer} [chars] - a buffer of input
+* @throws {Error} if not provided a buffer or null
+*/
+function incrparse( chars ) {
+	if ( isBuffer( chars ) ) {
+		this.buffer = this.buffer.concat( chars.toString( this.ENCODING ).split( '' ));
+	} else if ( chars === null ) {
+		this.buffer.push( null );
+	} else	{
+		throw new Error( 'provide a buffer or null' );
+	}
+	attemptParse.call( this );
+}
+setReadOnly( parseDsvStream.prototype, 'incrparse', incrparse);
```

Comment:
I think that, instead of `incrparse`, the method name can be `push`, which matches the Node.js stream convention (even if this is not a `stream`, nor should it be) and makes semantic sense. That we use `null` as the sentinel value also aligns with this parallel.